File: my-coffee-demo/src/main/java/com/example/mycoffeedemo/entity/Address.java
package com.example.mycoffeedemo.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("addresses")
public class Address {
    @TableId
    private Long id;
    private Long userId;
    private String recipientName;
    private String phone;
    private String province;
    private String city;
    private String district;
    private String detailAddress;
    private Boolean isDefault;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: my-coffee-vue/src/api/address.ts
import request from '@/utils/request'

export interface Address {
  id?: number
  userId?: number
  recipientName: string
  phone: string
  province?: string
  city?: string
  district?: string
  detailAddress?: string
  fullAddress?: string
  isDefault?: boolean
}

// 获取用户地址
export const getUserAddresses = (userId: number) =>
  request.get(`/address/${userId}`)

// 添加地址
export const addAddress = (address: Address) =>
  request.post('/address/add', address)

// 更新地址
export const updateAddress = (address: Address) =>
  request.put('/address/update', address)

// 删除地址
export const deleteAddress = (id: number) =>
  request.delete(`/address/delete/${id}`)

// 设置默认地址
export const setDefaultAddress = (id: number) =>
  request.put(`/address/${id}/default`)
    